src/Module/Maintenance.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Friendica\Module;
+
+use Friendica\BaseModule;
+use Friendica\Core\L10n;
+use Friendica\Core\Renderer;
+use Friendica\Core\System;
+use Friendica\Util\Strings;
+
+/**
+ * Shows the maintenance reason
+ * or redirects to the alternate location
+ */
+class Maintenance extends BaseModule
+{
+	public static function content()
+	{
+		$config = self::getApp()->getConfig();
+
+		$reason = $config->get('system', 'maintenance_reason');
+
+		if ((substr(Strings::normaliseLink($reason), 0, 7) === 'http://') ||
+			(substr(Strings::normaliseLink($reason), 0, 8) === 'https://')) {
+			System::externalRedirect($reason, 307);
+		}
+
+		header('HTTP/1.1 503 Service Temporarily Unavailable');
```

Comment:
You can replace the whole block below by
```php
$e = new HTTPException\ServiceUnavaiableException($reason);
$e->httpdesc = L10n::t('System down for maintenance');
throw $e;
```

Additionally, please rename the `ServiceUnavaiableException` to `ServiceUnavailableException`.

Reply:
Are you sure that this is working? I comited it this way.

I added for `system.maintenance_reason` the value "Good reason!" and set `system.maintenance` to true, but I always get this screen:
![Screenshot_2019-05-04 Friendica Social Network Maintenance](https://user-images.githubusercontent.com/379654/57180467-1d243d80-6e89-11e9-80e7-37943e13ab19.png)

In the source code, I see the whole friendica-structure, so it is loading something..

Reply:
There are two issues:
1. In `src/Module/Special/HTTPException.php` line 55, it should be `$message`, not `$description`.
2. `HTTPException->httpdesc` is entirely disregarded in `Module\Special\HTTPException->content()`.

Fixing the first issue would show the reason, but the error page title will still be the default one for this kind of error until we take `HTTPException->httpdesc` into account. Problem is this variable isn't always translated. It's initialized with the human-readable version of the exception english name, and the `HTTPException` module makes sure the title is actually a translated string.

I'm not sure how to proceed to allow default untranslated titles, defaults translated titles and custom titles.

Reply:
Why not something like:
```php
if (empty($this->httpdesc)) {
    $title = defaults($titles, $e->getCode(), 'Error ' . $e->getCode());
} else {
    $title = $this->httpdesc;
}
```

Reply:
This is the result of the new change:

![Screenshot_2019-05-04 Friendica Social Network Maintenance(1)](https://user-images.githubusercontent.com/379654/57183411-114a7280-6ead-11e9-9912-467afb43d213.png)

Reply:
> Why not something like:

Because `$exception->httpdesc` is set in the parent `HTTPExeption` constructor so the variable is always set, either with an untranslated title or the custom title.

Reply:
Ah, now I got it..